src/content/reference/fleet-management/migration_to_cluster_api.md

```diff
@@ -0,0 +1,44 @@
+
+## Pre-migration requirements
+
+Before you begin the migration:
+
+1. Your cluster should be on a release version `>=20.0.0`.
+2. The AWS IAM role, with the specific name `giantswarm-{CAPI_MC_NAME}-capa-controller`, must be created for the workload cluster's (WC) AWS account before starting the migration. [For more information please refer to this guide](https://docs.giantswarm.io/vintage/getting-started/cloud-provider-accounts/cluster-api/aws/#overview).
+
+__Note:__ The `CAPI_MC_NAME` is the name of the management cluster (MC) where the Cluster API controllers are installed.
+
+## Recommendations for a smooth migration
+
+We also recommend increasing the size of your "master" node instance type to 2x or 3x its normal size for the duration of the migration. (e.g. `mx.large` to `mx.4xlarge`). This ensures the API server can effectively handle the load during the migration since there might only be one node to handle the traffic at certain points throughout the process.
+
+## The migration process
+
+The migration process consists of several steps:
+
+1. __Initialization:__ Necessary Kubernetes access credentials and AWS credentials are retrieved.  Vault client is created to interact with the Vault instance containing all security assets of the cluster.
+2. __Preparation:__ Migration of secrets to the CAPI management cluster, including CA certs, encryption provider secrets, and service account secrets. Migration scripts are created as secrets in the CAPI management cluster. Additionally, AWS credentials for the cluster are migrated by creating an `AWSClusterRoleIdentity` in the CAPI management cluster. Certain operations are performed to avoid conflicts during migration, such as disabling machine health check on the vintage cluster resources, scaling down the app operator for the migrated workload cluster, or cleaning up certain charts.
+3. __Stopping vintage cluster resource reconciliation:__ To avoid conflicts, vintage reconciliation is stopped by removing all `aws-operator` labels from the vintage cluster resource.
+4. __Cluster API cluster provisioning:__ Generation and application of CAPI cluster templates. A separate routine runs in the background to ensure the old load balancer remains active. The tool waits until at least one CAPI control-plane node joins the cluster and is in a Ready state. Various operations are performed to ensure a smooth transition, such as stopping control-plane components on the vintage cluster, cordoning all vintage control-planes, and deleting certain pods to speed up installation and updates.
+5. __Cleaning the vintage cluster:__ All vintage control-plane nodes are drained, vintage auto scaling groups are deleted, and all worker nodes for each node pool are drained and deleted.
```

Comment:
We should mention any expectations the customer can have regarding uptime

Reply:
I think the goal is to have the same level of service as you would expect in a normal `update` so no downtime.